dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/VelocityRequestWrapper.java

```diff
@@ -1,347 +1,149 @@
-import java.security.Principal;
-import java.util.Collection;
-import java.util.Enumeration;
-import java.util.Locale;
-import java.util.Map;
-import java.util.Set;
 
-import javax.servlet.AsyncContext;
-import javax.servlet.DispatcherType;
-import javax.servlet.RequestDispatcher;
+import java.util.Set;
 import javax.servlet.ServletContext;
 import javax.servlet.ServletException;
-import javax.servlet.ServletInputStream;
-import javax.servlet.ServletRequest;
-import javax.servlet.ServletResponse;
-import javax.servlet.http.*;
-
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
 import com.dotmarketing.util.Config;
 import com.dotmarketing.util.SecurityLogger;
 import com.dotmarketing.util.UtilMethods;
 import com.google.common.collect.ImmutableSet;
 import com.liferay.portal.util.WebKeys;
 import com.liferay.util.Xss;
 
-public class VelocityRequestWrapper extends javax.servlet.http.HttpServletRequestWrapper  {
+/**
+ * This class wraps the incoming HttpServletRequest to provide security and to allow uris to be
+ * overridden
+ * 
+ * @author will
+ *
+ */
+public class VelocityRequestWrapper extends javax.servlet.http.HttpServletRequestWrapper {
+
 
-	private HttpServletRequest _request;
     private String customUserAgentHeader;
-	private String dotCMSUri;
-    final protected static Set<String> SET_VALUE_BLACKLIST = Config.getBooleanProperty("VELOCITY_PREVENT_SETTING_USER_ID", true) ? ImmutableSet.of(WebKeys.USER_ID, WebKeys.USER) : ImmutableSet.of();
-	public VelocityRequestWrapper(HttpServletRequest req) {
+    private String dotCMSUri;
+    final protected static Set<String> SET_VALUE_BLACKLIST =
+                    Config.getBooleanProperty("VELOCITY_PREVENT_SETTING_USER_ID", true)
+                                    ? ImmutableSet.of(WebKeys.USER_ID, WebKeys.USER)
+                                    : ImmutableSet.of();
+
+    private VelocityRequestWrapper(final HttpServletRequest req) {
         super(req);
-		this._request = req;
-	}
-	
-	public String getActualParameter(String arg0) {
-		return _request.getParameter(arg0);
-	}
-	
-	public String getAuthType() {
-		return _request.getAuthType();
-	}
 
-	public String getContextPath() {
-		return null;
-	}
+    }
 
-	public Cookie[] getCookies() {
-		return _request.getCookies();
-	}
+    public String getActualParameter(final String param) {
+        return super.getParameter(param);
+    }
 
-	public long getDateHeader(String arg0) {
-		return _request.getDateHeader(arg0);
-	}
+    @Override
+    public String getContextPath() {
+        return null;
+    }
 
-    public String getHeader ( String arg0 ) {
+    @Override
+    public String getHeader(final String header) {
 
-        if ( arg0 != null
-                && arg0.toLowerCase().equals( "user-agent" )
-                && this.getCustomUserAgentHeader() != null ) {
+        if ("user-agent".equalsIgnoreCase(header) && this.getCustomUserAgentHeader() != null) {
```

Comment:
Codacy found an issue: [Avoid using Literals in Conditional Statements](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5831320)